CMS-Enterprise/mint-app
# [MINT-3198] MTO table

# MINT-3198

## Description


- Added query for MTO Matrix
- Added utils for homogenize data
  - To work cleanly with the varying types/dataset, I have reformatted the query data so `MTOCategory`, `MTOSubcategory`, and `MTOMilestone` all have the same schema/data sctructure.
- Added `<ActionMenu />` for row dropdown
- Added util for manual/accessible reorder
- Added localStorage for toggle expanded rows as well as page length
- Added mutation and error messaging for mutation reorder
- Added unit tests

Instead of leveraging pagination to slice the data, under the hood the data is not being sliced, but rather not rendered.  `getRenderedRowIndexes` is a util that gets the map of indexes that need to be rendered.  This allows the data to be ordered even if items are hidden by page length/pagination.  Having the full dataset for sorting and pagination was required for the implementation with reordering of categories and subcategories. 

Commented out some code in case we use it for reordering milestones


## How to test this change

1.  Seed data
2. Navigate to MTO seeded plan and to MTO Matrix part of app
3. Verify Data populates
4. Verify drag drop (subcategories can be moved across categories)
5. Verify row menu works to move items up and down
6. Verify pagination and pagelength
7. Verify table sort
8. Verify local storage for expanded rows and page length

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
